Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -55,9 +57,13 @@ class CallUIAdapter {
             Logger.info("\(TAG) choosing non-callkit adaptee for older iOS")
             adaptee = NonCallKitCallUIAdaptee(callService: callService, notificationsAdapter: notificationsAdapter)
         }
+
+        audioService = CallAudioService(handleRinging: adaptee.hasManualRinger)
     }
 
     internal func reportIncomingCall(_ call: SignalCall, thread: TSContactThread) {
+        call.addObserverAndSyncState(observer: audioService)
```

Comment:
I'd say it's the CallService's responsibility to ensure that `CallUIAdapter.reportIncomingCall` is only called when the app is "ready" to report an incoming call (e.g. it there's not already another call in-progress).

As far as the the ringer being in a "bad state", I don't *think* that's possible, as the call service coordinates that we're not reportingIncomingCall for multiple calls, and the audioService is intended to be idempotent WRT the signalCall's state changes.